CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+project(igrec)
+cmake_minimum_required(VERSION 2.8)
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "RelWithAsserts" CACHE STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+
+add_subdirectory(configs)
+add_subdirectory(src)
+
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  set(CMAKE_INSTALL_PREFIX "/usr/local" CACHE PATH "default install path" FORCE)
+endif()
+
+# Expand tildas and convert to absolute path
+FILE(TO_CMAKE_PATH "${CMAKE_INSTALL_PREFIX}" CMAKE_INSTALL_PREFIX)
+
+set(INSTALL_DIR "share/igrec")
```

Comment:
For linux consider this:
http://refspecs.linuxfoundation.org/FHS_2.3/fhs-2.3.html#OPTADDONAPPLICATIONSOFTWAREPACKAGES
and 
http://unix.stackexchange.com/questions/20600/should-i-put-application-into-usr-local-or-usr-local-share
Why not /opt?

Moreover, http://refspecs.linuxfoundation.org/FHS_2.3/fhs-2.3.html#USRLOCALLOCALHIERARCHY
states that /usr/local is probably not what we want.

On Mac OS (as a BSD)`man her`gives me this: `/usr/local` -- executables, libraries, etc. not included by the basic operating system. `/opt` is not mentioned there btw.

Also consider the first answer [here](http://apple.stackexchange.com/questions/119230/what-is-standard-for-os-x-filesystem-e-g-opt-vs-usr).

Therefore, I suggest /opt on Linux and /usr/local on BSD.